### ISSUE TITLE
[Design] Automated tests

### ISSUE THREAD

**Issue by tuliomir** [HathorNetwork/hathor-wallet#357]
:construction: Under development

This issue describes the research being done on automating tests for the Wallet Desktop application.

**Comment by tuliomir** [comment]
Reviewers: @pedroferreira1 , @alexruzenhack

**Comment by alexruzenhack** [comment]
> It is possible to test UI without Puppeteer, we can use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) + Jest. Did you hear about this library? With this combo, the components can be rendered and navigated by DOM API and then asserted.

Now I'm reviewing the PoC and see you use the React Testing Library already. Nice!

**Comment by pedroferreira1** [comment]
> As such, possible paths for this UI test would be:
>
> Update all react dependencies to use react-scripts@5 and try again

We've talked about trying to upgrade only the react-scripts and build a poc of one test suite using pupeteer, even if some parts of the app end up broken. Did you try to do it?

**Comment by tuliomir** [comment]
> ...we can use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) + Jest.

Added a _Guide Level Description_ section documenting that these libs are being used on the PoC. :+1:

**Comment by pedroferreira1** [comment]
> To avoid unnecessary consumption of HTR tokens, all tests should, at the end, melt every custom token generated to ensure the original HTR balance will be preserved - exception made for the NFTs that have a necessary fee. Those funds should finally be sent back to the source wallet to provide for future tests.

Why don't we run the tests using the testnet, so we don't really need to worry about losing funds?

**Comment by tuliomir** [comment]
> ...have you validated how to run those tests using ledger? 

Added considerations about the hardware wallet / ledger testing: they will be done in a future moment and are described on the _Future Possibilities_  section.

---

> ...we don't really need to worry about losing funds?

About the amount of funds, while we have a lot of funds available on the `testnet`, we also have to refill the funds on the source wallet manually. Since the tests are automated and may be executed very often, I think It's better to preserve these funds only to delay as much as possible the manual refilling process that will be done by the _on-call_.

What do you think?

---
Also, added a section on mocking and intercepting HTTP requests on the _Future Possibilities_ section to cover a comment on the PoC PR.

**Comment by pedroferreira1** [comment]
> About the amount of funds, while we have a lot of funds available on the testnet, we also have to refill the funds on the source wallet manually. Since the tests are automated and may be executed very often, I think It's better to preserve these funds only to delay as much as possible the manual refilling process that will be done by the on-call.
> 
> What do you think?

Yeah, this is good, I just think we should spend much time on that, especially when you say

> It also would make it easier to change the values on some of the current QA tests to a fraction of what they are, so that fewer tokens are needed for each test iteration.

When we create a token, we need to create at least 1.00 amount to be able to melt and get the HTR deposit back. And even returning the tokens, it's always better to use testnet token and this was not mentioned in the design, that's why I wrote about it.
